### PR TITLE
Add Preview version SDK(2022-10-31-preview), CLI Credential Integration Example

### PR DESCRIPTION
**Pull request by yang-hai-feng**
…h Azure CLI Credentials

### CODE REVIEW

File: samples/PreviewExamples/Program.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+using Azure.Identity;
+using System;
+
+namespace PreviewExamples {
+    internal class Program
+    {        
+        private static void Main(string[] args)
+        {
+            // Using 'az login' first, and using 'az account set -s subscription'
+            var credential = new AzureCliCredential();
```
Comment (asanjabi) [inline samples/PreviewExamples/Program.cs]: Would DefaultAzureCredential work here?

File: samples/PreviewExamples/DeviceExample.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    public class DeviceExample {
+        public static void Run(string subdomain, TokenCredential credential) {
+            var devicesClient = new DevicesClient(subdomain, credential);
+
+            // Demo new functions for filter, paging, orderby
+
+            // Create list of edge Devices
+            for (int i = 0; i < 5; i++) {
+                var deviceId = $"testdevice_{i}";
+                if (devicesClient.Get(deviceId).GetRawResponse().Status == 404) {
```
Comment (asanjabi) [inline samples/PreviewExamples/DeviceExample.cs]: The new SDK throws a RequestFailedException with status set to 404.

**Review by asanjabi — CHANGES_REQUESTED** [review]

File: samples/PreviewExamples/DeviceExample.cs (hunk 2 above)
Comment (yang-hai-feng) [reply samples/PreviewExamples/DeviceExample.cs]: Fixed.

File: samples/PreviewExamples/Program.cs (hunk 1 above)
Comment (yang-hai-feng) [reply samples/PreviewExamples/Program.cs]: Tried it was working as well, but it might also because I already did CLI login from my console.

**Review by asanjabi — APPROVED** [review]

**Pull request merged**